update the L1 and HLT post-LS1 customisations

fwyzard opened the pull request:
- add customisations to use the 25ns v0 and 50ns v0 L1 menus
- separate the L1 customisations for 50ns, 25ns and heavy ions
- move L1 HI menu customisation from L1Trigger_custom.py to customise_overwriteL1Menu.py
- update SLHCUpgradeSimulations/Configuration/python/postLS1Customs.py

fwyzard committed:
update the L1 post-LS1 customisations

  - add customisations to use the 25ns v0 and 50ns v0 L1 menus
  - separate the L1 customisations for 50ns, 25ns and heavy ions
  - move L1 HI menu customisation from L1Trigger_custom.py to customise_overwriteL1Menu.py
  - update SLHCUpgradeSimulations/Configuration/python/postLS1Customs.py

cmsbuild commented:
A new Pull Request was created by @fwyzard (Andrea Bocci) for CMSSW_7_4_X.

update the L1 post-LS1 related customisations

It involves the following packages:

L1Trigger/Configuration
L1Trigger/L1TCommon
SLHCUpgradeSimulations/Configuration


@cmsbuild, @civanch, @nclopezo, @mulhearn, @mdhildreth can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented on SLHCUpgradeSimulations/Configuration/python/postLS1Customs.py:
```diff
@@ -90,145 +88,8 @@ def customiseRun2EraExtras(process):
         process.simCsctfTrackDigis.SectorReceiverInput = cms.untracked.InputTag("simCscTriggerPrimitiveDigis", "MPCSORTED")
 
     # deal with L1 Emulation separately:
-    # replace the L1 menu from the global tag with one of the following alternatives
```
Note that here I've replaced the copy of the L1 customisations with a call to the original customisation function: the copy had become out-of-sync from the original, and I hardly see the point of maintaining the same customisation in two places.

fwyzard commented on SLHCUpgradeSimulations/Configuration/python/postLS1Customs.py:
```diff
@@ -1,19 +1,17 @@
-    customiseSimL1EmulatorForPostLS1(process)
+    from L1Trigger.L1TCommon.customsPostLS1 import customiseSimL1EmulatorForPostLS1_25ns
+    process = customiseSimL1EmulatorForPostLS1_25ns(process)
 
```
Note that I'm still calling the 25ns customisation unconditionally.

fwyzard commented on L1Trigger/L1TCommon/python/customsPostLS1.py:
```diff
@@ -135,30 +107,29 @@ def customiseSimL1EmulatorForPostLS1_nomenu(process):
+from L1Trigger.Configuration.customise_overwriteL1Menu import *
+
+def customiseSimL1EmulatorForPostLS1_50ns(process):
+    # move to the 50ns v0 menu once the HLT has been updated accordingly
```
Note that I'm still using the 25ns v2 L1 menu, as the HLT is not yet ready to use the dedicated 50ns v0 menu.

fwyzard commented:
@davidlange6 , so far here there should be no changes to any workflow.

Should I make a copy of postLS1Customs.py specifically for 50ns, to load the different L1 menu ?
What about Heavy Ions ?

Commit:
add specific post-LS1 customisations for 25ns, 50ns and heavy ions

fwyzard commented:
OK, I've added dedicated post-LS1 customisations for 25ns, 50ns and Heavy Ions.

However, they are not used yet: somebody that knows how would need to update Configuration/PyReleaseValidation/python/relval_steps.py to use them as appropriate.

cmsbuild commented:
Pull request #7579 was updated. @cmsbuild, @civanch, @nclopezo, @mulhearn, @mdhildreth can you please check and sign again.

davidlange6 commented:
we can handle this internally to the python instead. I'll take care

On Feb 5, 2015, at 6:48 PM, Andrea Bocci <notifications@github.com>
 wrote:

> OK, I've added dedicated post-LS1 customisations for 25ns, 50ns and Heavy Ions.
> 
> However, they are not used yet: somebody that knows how would need to update Configuration/PyReleaseValidation/python/relval_steps.py to use them as appropriate.
> 
> �
> Reply to this email directly or view it on GitHub.
>

civanch commented:
@fwyzard , @davidlange6, in the integrated PR #7503 HF shower library was moved from postLS1Customs.py to g4SimHits_cff.py and enabled via run2 era.

Is it optimal to have HF shower library customisation in 2 new py files for 25 and 50 ns? I mean should we keep or remove lines:
 
def customise_Sim(process):
   # enable 2015 HF shower library
   process.g4SimHits.HFShowerLibrary.FileName = 'SimG4CMS/Calo/data/HFShowerLibrary_npmt_noatt_eta4_16en_v3.root'
   return process

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

davidlange6 commented:
Hi Vladimir
I believe you have not read the code completely correctly and instead see changes related to starting the support for eras.

David


On Feb 5, 2015, at 8:05 PM, Vladimir Ivantchenko <notifications@github.com>
 wrote:

> @fwyzard , @davidlange6, in the integrated PR #7503 HF shower library was moved from postLS1Customs.py to g4SimHits_cff.py and enabled via run2 era.
> 
> Is it optimal to have HF shower library customisation in 2 new py files for 25 and 50 ns? I mean should we keep or remove lines:
> 
> def customise_Sim(process):
> # enable 2015 HF shower library
> process.g4SimHits.HFShowerLibrary.FileName = 'SimG4CMS/Calo/data/HFShowerLibrary_npmt_noatt_eta4_16en_v3.root'
> return process
> 
> �
> Reply to this email directly or view it on GitHub.
>

cmsbuild commented:
-1 
 
Tested at: 3c30e7878e273f63901ef02b69ac6ca3b1912fa8
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
---> test runtestTqafTopHitFit had ERRORS
---> test runtestTqafTopJetCombination had ERRORS
---> test runtestTqafTopEventProducers had ERRORS
---> test runtestTqafExamples had ERRORS
---> test runtestTqafTopKinFitter had ERRORS
---> test runtestPhysicsToolsPatAlgos had ERRORS
---> test testJetMETCorrectionsType1MET had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2407/summary.html

fwyzard commented:
Mhm, this is unexpected... I can double check tomorrow, but maybe the errors are unrelated ?

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2407/summary.html

cmsbuild commented:
-1 
 
Tested at: 9ca5cb1acfc634f8464abc722e57da03fe5297b9
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
---> test runtestTqafTopHitFit had ERRORS
---> test runtestPhysicsToolsPatAlgos had ERRORS
---> test runtestTqafExamples had ERRORS
---> test testJetMETCorrectionsType1MET had ERRORS
---> test runtestTqafTopKinFitter had ERRORS
---> test runtestTqafTopEventProducers had ERRORS
---> test runtestTqafTopJetCombination had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2411/summary.html

cmsbuild commented:
-1 
 
Tested at: 3c30e7878e273f63901ef02b69ac6ca3b1912fa8
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS
---> test testJetMETCorrectionsType1MET had ERRORS
---> test runtestTqafTopEventProducers had ERRORS
---> test runtestTqafTopHitFit had ERRORS
---> test runtestTqafExamples had ERRORS
---> test runtestTqafTopJetCombination had ERRORS
---> test runtestTqafTopEventSelection had ERRORS
---> test runtestTqafTopKinFitter had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2404/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2411/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2404/summary.html

Martin-Grunewald commented:
Hi
```
OK, I've added dedicated post-LS1 customisations for 25ns, 50ns and Heavy Ions.
```
that looks like a horrible duplication of (nearly) identical PostLS1 files.

fwyzard committed:
remove the HLT post-LS1 customisations

remove the HLT specific post-LS1 customisations, as they are already
integrated in the latest HLT menu dumps

cmsbuild commented:
Pull request #7579 was updated. @civanch, @nclopezo, @mdhildreth, @cmsbuild, @diguida, @cerminar, @ggovi, @mmusich, @mulhearn can you please check and sign again.

Martin-Grunewald commented:
Also, I still think that the three SLHCUpgradeSimulations/Configuration/python/postLS1Customs*
files for HI/25/50 have a huge overlap which may go out of sync... Ideally one would use
a single common one called by the three specific ones in addition to the three sets of
specific changes...

fwyzard commented:
I agree, but haven't had the time to merge them properly yet.

Ciao,
.Andrea
On 6 Feb 2015 15:58, "Martin Grunewald" <notifications@github.com> wrote:

> Also, I still think that the three
> SLHCUpgradeSimulations/Configuration/python/postLS1Customs*
> files for HI/25/50 have a huge overlap which may go out of sync... Ideally
> one would use
> a single common one called by the three specific ones in addition to the
> three sets of
> specific changes...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73248616>.
>

davidlange6 commented:
As this stands, this PR loads the L1 menus for all workflows whether or not they have anything to do with L1. I believe the steps that need customiseSimL1EmulatorForPostLS1_25ns/50ns are "L1", "HLT", and "L1Reco". Is that correct?

fwyzard commented:
Hi David,
you need the L1 menu also in DQM, and PAT, otherwise you read the wrong one
from the global tag.

I do not know if other workflows could be affected, too. Maybe AlCa ?

Ciao,
.Andrea
On 6 Feb 2015 18:19, "David Lange" <notifications@github.com> wrote:

> As this stands, this PR loads the L1 menus for all workflows whether or
> not they have anything to do with L1. I believe the steps that need
> customiseSimL1EmulatorForPostLS1_25ns/50ns are "L1", "HLT", and "L1Reco".
> Is that correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73275731>.
>

davidlange6 commented:
as an example, remind me how PAT uses the L1 menu directly? That would give me a good example to work from

On Feb 6, 2015, at 6:27 PM, Andrea Bocci <notifications@github.com>
 wrote:

> Hi David,
> you need the L1 menu also in DQM, and PAT, otherwise you read the wrong one
> from the global tag.
> 
> I do not know if other workflows could be affected, too. Maybe AlCa ?
> 
> Ciao,
> .Andrea
> On 6 Feb 2015 18:19, "David Lange" <notifications@github.com> wrote:
> 
> > As this stands, this PR loads the L1 menus for all workflows whether or
> > not they have anything to do with L1. I believe the steps that need
> > customiseSimL1EmulatorForPostLS1_25ns/50ns are "L1", "HLT", and "L1Reco".
> > Is that correct?
> >
> > �
> > Reply to this email directly or view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73275731>.
> >
> �
> Reply to this email directly or view it on GitHub.
>

fwyzard commented:
Hi David,
I don't know by heart, I remember noticing that the PAT step accesses the
L1 menu, while investigating which step uses which conditions.

I guess it's to map the L1 trigger bits to trigger names.

Ciao,
.Andrea
as an example, remind me how PAT uses the L1 menu directly? That would give
me a good example to work from

On Feb 6, 2015, at 6:27 PM, Andrea Bocci <notifications@github.com>
wrote:

> Hi David,
> you need the L1 menu also in DQM, and PAT, otherwise you read the wrong
one
> from the global tag.
>
> I do not know if other workflows could be affected, too. Maybe AlCa ?
>
> Ciao,
> .Andrea
> On 6 Feb 2015 18:19, "David Lange" <notifications@github.com> wrote:
>
> > As this stands, this PR loads the L1 menus for all workflows whether or
> > not they have anything to do with L1. I believe the steps that need
> > customiseSimL1EmulatorForPostLS1_25ns/50ns are "L1", "HLT", and
"L1Reco".
> > Is that correct?
> >
> > �
> > Reply to this email directly or view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73275731>.
> >
> �
> Reply to this email directly or view it on GitHub.
>

—
Reply to this email directly or view it on GitHub
<https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73280341>.

cmsbuild commented:
-1 
Tested at: 5b7cbce48a0e9b9e59356c3fe3cd9e2654890a45
 When I ran the RelVals I found an error in the following worklfows: 
 4.22 step2
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step2_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

----- Begin Fatal Exception 06-Feb-2015 20:37:15 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 160960

   [1] Running path 'reconstruction_step'

   [2] Calling beginRun for module CtfSpecialSeedGenerator/'combinatorialcosmicseedfinderP5'

   [3] Using EventSetup component TkTransientTrackingRecHitBuilderESProducer/'ttrhbwr' to make data TransientTrackingRecHitBuilder/'WithTrackAngle' in record TransientRecHitRecord

   [4] Using EventSetup component StripCPEESProducer/'StripCPEfromTrackAngleESProducer' to make data StripClusterParameterEstimator/'StripCPEfromTrackAngle' in record TkStripCPERecord

   [5] Using EventSetup component TrackerDigiGeometryESModule/'trackerGeometryDB' to make data TrackerGeometry/'' in record TrackerDigiGeometryRecord

   [6] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record TrackerAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

----- Begin Fatal Exception 06-Feb-2015 20:37:39 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 182124

   [1] Calling global beginRun for module DTTrigProd/'valDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

----- Begin Fatal Exception 06-Feb-2015 20:37:48 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Running path 'simulation_step'

   [2] Calling beginRun for module FamosProducer/'famosSimHits'

   [3] Using EventSetup component TrackerDigiGeometryESModule/'misalignedTrackerGeometry' to make data TrackerGeometry/'MisAligned' in record TrackerDigiGeometryRecord

   [4] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record TrackerAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

----- Begin Fatal Exception 06-Feb-2015 20:38:04 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

----- Begin Fatal Exception 06-Feb-2015 20:38:14 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Running path 'simulation_step'

   [2] Calling beginRun for module FamosProducer/'famosSimHits'

   [3] Using EventSetup component TrackerDigiGeometryESModule/'misalignedTrackerGeometry' to make data TrackerGeometry/'MisAligned' in record TrackerDigiGeometryRecord

   [4] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record TrackerAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT+MINIAODDreHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT+MINIAODDreHLT.log

----- Begin Fatal Exception 06-Feb-2015 20:38:23 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 194533

   [1] Calling global beginRun for module DTTrigProd/'valDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 06-Feb-2015 20:39:22 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
101.0 step1
<pre>runTheMatrix-results/101.0_SingleElectronE120EHCAL+SingleElectronE120EHCAL/step1_SingleElectronE120EHCAL+SingleElectronE120EHCAL.log

----- Begin Fatal Exception 06-Feb-2015 20:39:30 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1 lumi: 1 event: 1

   [1] Running path 'local_digireco'

   [2] Calling event method for module MixingModule/'mix'

   [3] Using EventSetup component TrackerDigiGeometryESModule/'trackerGeometryDB' to make data TrackerGeometry/'' in record TrackerDigiGeometryRecord

   [4] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record TrackerAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

----- Begin Fatal Exception 06-Feb-2015 20:39:56 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 165121

   [1] Calling global beginRun for module DTTrigProd/'valDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD.log

----- Begin Fatal Exception 06-Feb-2015 20:40:02 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 165121

   [1] Calling global beginRun for module DTTrigProd/'valDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

----- Begin Fatal Exception 06-Feb-2015 20:40:59 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 191226

   [1] Calling global beginRun for module DTTrigProd/'valDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 06-Feb-2015 20:43:45 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

----- Begin Fatal Exception 06-Feb-2015 20:44:20 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECO+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECO+HARVEST+ALCATT.log

----- Begin Fatal Exception 06-Feb-2015 20:45:19 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 06-Feb-2015 20:48:39 CET-----------------------

An exception of category 'StdException' occurred while

   [0] Processing run: 1

   [1] Calling global beginRun for module DTTrigProd/'simDtTriggerPrimitiveDigis'

   [2] Using EventSetup component DTGeometryESModule/'' to make data DTGeometry/'' in record MuonGeometryRecord

   [3] Using EventSetup component PoolDBESSource/'GlobalTag' to make data AlignmentErrorsExtended/'' in record DTAlignmentErrorExtendedRcd

Exception Message:

A std::exception was thrown.

unsupported version

----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2423/summary.html

davidlange6 commented:
Ok - i'll see what I can learn.

On Feb 6, 2015, at 7:13 PM, Andrea Bocci <notifications@github.com>
 wrote:

> Hi David,
> I don't know by heart, I remember noticing that the PAT step accesses the
> L1 menu, while investigating which step uses which conditions.
> 
> I guess it's to map the L1 trigger bits to trigger names.
> 
> Ciao,
> .Andrea
> as an example, remind me how PAT uses the L1 menu directly? That would give
> me a good example to work from
> 
> On Feb 6, 2015, at 6:27 PM, Andrea Bocci <notifications@github.com>
> wrote:
> 
> > Hi David,
> > you need the L1 menu also in DQM, and PAT, otherwise you read the wrong
> one
> > from the global tag.
> >
> > I do not know if other workflows could be affected, too. Maybe AlCa ?
> >
> > Ciao,
> > .Andrea
> > On 6 Feb 2015 18:19, "David Lange" <notifications@github.com> wrote:
> >
> > > As this stands, this PR loads the L1 menus for all workflows whether or
> > > not they have anything to do with L1. I believe the steps that need
> > > customiseSimL1EmulatorForPostLS1_25ns/50ns are "L1", "HLT", and
> "L1Reco".
> > > Is that correct?
> > >
> > > �
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73275731>.
> > >
> > �
> > Reply to this email directly or view it on GitHub.
> >
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73280341>.
> —
> Reply to this email directly or view it on GitHub.
>

Martin-Grunewald commented:
PAT trigger code uses the HLTConfigProvider and also L1GtUtils to access L1
names which needs the proper L1 menu.
Any code using the HLTConfigProvider (DQM etc.) may access
L1 menu information incl. L1 trigger names.

davidlange6 commented:
ie, all cms analysis workflows possibly need the menu. ok - it needs to be back in the GT asap� not sure how to support this in the meanwhile, but will find something.

On Feb 7, 2015, at 7:42 AM, Martin Grunewald <notifications@github.com>
 wrote:

> PAT trigger code uses the HLTConfigProvider and also L1GtUtils to access L1
> names which needs the proper L1 menu.
> Any code using the HLTConfigProvider (DQM etc.) may access
> L1 menu information incl. L1 trigger names.
> 
> �
> Reply to this email directly or view it on GitHub.
>

fwyzard commented:
Rebased, and cleaned up the different cutsomisations.

This is my best bet at automatically picking up the correct customisation for 50ns vs. 25ns - but will likely fail for workflows that run the L1 emulator without running the mixing module.

Heavy Ions needs to be enabled explicitly.

Actually, the best solution would be to enable any of the three specific scenarios explicitly via cmsDriver.

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #7579 was updated. @cmsbuild, @civanch, @nclopezo, @mulhearn, @mdhildreth can you please check and sign again.

cmsbuild commented:
-1 
Tested at: 300e7a326e5ed25fe049f37d72c372502941c281
 When I ran the RelVals I found an error in the following worklfows: 
 1306.0 step1
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step1_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 09-Feb-2015 06:19:46 CET-----------------------

An exception of category 'ConfigFileReadError' occurred while

   [0] Processing the python configuration file named SingleMuPt1_pythia8_cfi_GEN_SIM.py

Exception Message:

python encountered the error: <type 'exceptions.AttributeError'>

'NoneType' object has no attribute 'fillProcessDesc'

----- End Fatal Exception -------------------------------------------------

</pre>
1330.0 step1
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step1_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 09-Feb-2015 06:19:48 CET-----------------------

An exception of category 'ConfigFileReadError' occurred while

   [0] Processing the python configuration file named ZMM_13TeV_TuneCUETP8M1_cfi_GEN_SIM.py

Exception Message:

python encountered the error: <type 'exceptions.AttributeError'>

'NoneType' object has no attribute 'fillProcessDesc'

----- End Fatal Exception -------------------------------------------------

</pre>
25202.0 step1
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15/step1_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 09-Feb-2015 06:19:52 CET-----------------------

An exception of category 'ConfigFileReadError' occurred while

   [0] Processing the python configuration file named TTbar_13TeV_TuneCUETP8M1_cfi_GEN_SIM.py

Exception Message:

python encountered the error: <type 'exceptions.AttributeError'>

'NoneType' object has no attribute 'fillProcessDesc'

----- End Fatal Exception -------------------------------------------------

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2450/summary.html

davidlange6 commented:
I would suggest we just key the menu off of the global tag instead of trying to figure out the right pileup config for every workflow. This is in any case where the menus need to get back to.

fwyzard commented:
- rebased
  - reverted the heavy ions L1 menu to the 25ns menu for the moment (until the relevant HLT is updated)
  - propagate the alternative customisePostLS1 to the matrix, leaving the 25ns case as the default

fwyzard commented:
@cmsbuld please test

fwyzard commented:
`runTheMatrix.py -l limited` seems to work for me, on top of CMSSW_7_4_X_2015-02-09-1400

cmsbuild commented:
Pull request #7579 was updated. @civanch, @nclopezo, @boudoul, @franzoni, @mdhildreth, @cmsbuild, @srimanob, @mulhearn can you please check and sign again.

cmsbuild commented:
+1
Tested at: 206aa4791fb23e5356e99819c6c5a38c6b38ec3a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2484/summary.html

franzoni commented:
hello, I've taken a first look and intend to digest further.

Are we sure we want to go down this path ?
The complexity of interplay between the underlying 25ns default and the "tweak on top" actions is beyond what we can reliably control.  Kudos to @fwyzard et all who have put energies into this. 
I feel uncomfortable and that we're cooking up an unmaintainable system, am I the only one.

...when postLS1Customs was introduced in 2013 and put in production and relval,  the reassuring comment to those who pointed out it was not a good direction to go was: that it would be temporary and soon replaced by a more durable solution. :-) 

Before scrutinising the details of the relval_step I'll want to think more.

fwyzard commented:
Hi, FWIW I agree with Giovanni about the maintainability of this approach.

Here, I simply wanted to make the 50ns/25ns/HI cases explicit in the matrix in order to allow different L1 emulator configuration and menu. The splitting of the post-LS1 customisation is kind of a side effect.

Is there a way to expand the python files for all the workflows in the matrix ?
That could be a way to make sure that no (unwanted) changes enter any workflow.

davidlange6 commented:
On Feb 10, 2015, at 10:17 AM, Andrea Bocci <notifications@github.com>
 wrote:

> Hi, FWIW I agree with Giovanni about the maintainability of this approach.
> 
> Here, I simply wanted to make the 50ns/25ns/HI cases explicit in the matrix in order to allow different L1 emulator configuration and menu. The splitting of the post-LS1 customisation is kind of a side effect.
> 
> Is there a way to expand the python files for all the workflows in the matrix ?
> That could be a way to make sure that no (unwanted) changes enter any workflow.
> 

you can call dumpPython on the config that gets created (e.g., file.write(process.dumpPython) )



> �
> Reply to this email directly or view it on GitHub.
>

davidlange6 commented:
Lets be clear that the complication here is that the "right" L1 menu can not be deduced simply in all cases, as it is needed by essentially all CMS workflows. This means that short of a user specifying that her data is 25ns or 50ns, there is no good solution aside from a GT driven one (so "eras" will suffer from the same need to specify enough information for 25/50) or by file providence (which means have an input file accessible). Other use cases are simpler as they can be derived from knowing that a "run 2" configuration is wanted and from the process PSet itself. 

The menu has been in the GT and people have written code assuming that it was in the GT - the current discussion is driven by the (hopefully) temporary move of the menu out of the DB.  


On Feb 10, 2015, at 10:13 AM, Giovanni Franzoni <notifications@github.com>
 wrote:

> hello, I've taken a first look and intend to digest further.
> 
> Are we sure we want to go down this path ?
> The complexity of interplay between the underlying 25ns default and the "tweak on top" actions is beyond what we can reliably control. Kudos to @fwyzard et all who have put energies into this. 
> I feel uncomfortable and that we're cooking up an unmaintainable system, am I the only one.
> 
> ...when postLS1Customs was introduced in 2013 and put in production and relval, the reassuring comment to those who pointed out it was not a good direction to go was: that it would be temporary and soon replaced by a more durable solution. :-)
> 
> Before scrutinising the details of the relval_step I'll want to think more.
> 
> �
> Reply to this email directly or view it on GitHub.
>

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2484/summary.html

civanch commented:
+1

fwyzard commented:
On 10 February 2015 at 10:51, David Lange <notifications@github.com> wrote:

> you can call dumpPython on the config that gets created (e.g.,
> file.write(process.dumpPython) )



well,
```
runTheMatrix --dryRun
```
does not create the python files at all, so I cannot easily dump them :-(

.A

davidlange6 commented:
its runTheMatrix --no_exec if I remember correctly


On Feb 10, 2015, at 12:06 PM, Andrea Bocci <notifications@github.com>
 wrote:

> On 10 February 2015 at 10:51, David Lange <notifications@github.com> wrote:
> 
> > you can call dumpPython on the config that gets created (e.g.,
> > file.write(process.dumpPython) )
> 
> 
> 
> well,
> ```
> runTheMatrix --dryRun
> ```
> does not create the python files at all, so I cannot easily dump them :-(
> 
> .A
> �
> Reply to this email directly or view it on GitHub.
>

mark-grimes commented:
Still digesting this, but on the point of dumping config files:
`--no_exec` doesn't work and I don't see anything similar in the runTheMatrix source file (I could have missed it though).
I've always used

    runTheMatrix -l <something> --dryRun
    cd <something>*
    sed s:">":"--no_exec #": cmdLog | sh

(adds `--no_exec`, comments out the redirections then executes the cmsDriver command)
I guess if you're doing everything (no `-l`) you'll need something like

    runTheMatrix --dryRun
    ls | awk '{print "cd "$0"; sed s:\">\":\"--no_exec #\": cmdLog | sh; cd .."}' | sh

Bit of a ballache but it works.  Then when I was having to check for the eras I used

    ls */*.py | awk -F "\\\\.py" '{print "edmConfigDump "$0" > "$1"_dump.py"}' | sh

(The `\\\\` stops awk interpreting `.` as "any character" and matching against e.g. `_pythia8`)
Note that any print statements in the configuration tree are also added to the output and can stop the dumped files from being python parseable, but this works well for a simple diff.

fwyzard commented:
Hi Mark,

On 10 February 2015 at 22:04, Mark Grimes <notifications@github.com> wrote:

> --no_exec doesn't work and I don't see anything similar in the
> runTheMatrix source file (I could have missed it though).
>

I've used

runTheMatrix.py --command=--no_exec

to create all .py files, and then

find * -name '*.py' | xargs -n1 -i -P16 sh -c 'edmConfigDump "$1" >
"$1"_dump.py' -- {}

to expand them all.

-P16 runs 16 jobs in parallel, which is handy if you have enough cores.


The diff of all expanded dumps with and without my changes counted over 30k
lines, so I've given up checking it after a while, and encounterng only
expected (or anyway plausible) changes.

.A

Martin-Grunewald commented:
Whoa hold on. ```--no_exec``` must be kept working, we run our tests this way!

Martin-Grunewald commented:
This PR now has merge conflicts....

fwyzard commented:
@Martin-Grunewald I did not touch the `--no_exec` functionality, I'm just reporting how to use it with `runTheMatrix.py`

fwyzard commented:
rebased on top of the latest CMSSW_7_4_X

cmsbuild commented:
Pull request #7579 was updated. @civanch, @nclopezo, @boudoul, @franzoni, @mdhildreth, @cmsbuild, @srimanob, @mulhearn can you please check and sign again.

slava77 commented:
I'm usually using "-j 0" (I think this way of making --no_exec predates
appearance of --command)
does this still work?



On 2/11/15 2:42 AM, Andrea Bocci wrote:
> @Martin-Grunewald <https://github.com/Martin-Grunewald> I did not touch
> the |--no_exec| functionality, I'm just reporting how to use it with
> |runTheMatrix.py|
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7579#issuecomment-73848652>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 TAMU: Physics Dept Texas A&M MS4242, College Station, TX 77843-4242
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 1ca90725ba6342c01d4d797edf49e45ac90feeb3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2522/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7579/2522/summary.html

davidlange6 commented:
bypassing since we already agreed on this PR (and its comparisons are all ok)

franzoni commented:
hi physics-week => missed the train 
:-(

I believe that this: digiPremixUp2015Defaults50ns 
should also receive the 50ns specific customisation.
@mdhildreth can you also cross check, please ?

@fwyzard : what effect do we expect on HLT performance in pre7; it would be good if you / Darren / Roberta could spell out the impact to be explained to the validators

These are less important aspects:
step3Up2015Hal  => should be customised as 25 ns ?
HARVESTGEN => no need of any customisation

fwyzard commented:
On 12 February 2015 at 19:09, Giovanni Franzoni <notifications@github.com>
wrote:

> I believe that this: digiPremixUp2015Defaults50ns
> should also receive the 50ns specific customisation.
> @mdhildreth <https://github.com/mdhildreth> can you also cross check,
> please ?
>
I guess so, from the name, but what is that ?
Something in the matrix ?

@fwyzard <https://github.com/fwyzard> : what effect do we expect on HLT
> performance in pre7; it would be good if you / Darren / Roberta could spell
> out the impact to be explained to the validators
>
No L1 or HLT changes to the 25ns amples from this PR (though of course
there may be changes from the other L1 or HLT-ralated PR's).

The 50ns samples should be properly reverted to the Legacy L1 emulator, so
they should observe changes to most calorimetric related L1 quantities
(Jets, MET, E/Gamma, Taus). Do we actually have 50ns relvals ?

The HI samples should properly use the HI-specific settings for the Stage 1
L1 emulator, but I do not know what that means in practice (I would guess a
different treatment of the pileup, but I do not really know).

These are less important aspects:
> step3Up2015Hal => should be customised as 25 ns ?
> HARVESTGEN => no need of any customisation
>
.A

franzoni commented:
see #7826 for follow up about premixing workflows
